lib/output.js

```diff
@@ -74,9 +74,12 @@ function OutputStream(options) {
     var OUTPUT = "";
 
     function to_ascii(str, identifier) {
-        return str.replace(/[\u0000-\u001f\u007f-\uffff]/g, function(ch) {
-            var code = ch.charCodeAt(0).toString(16);
-            if (code.length <= 2 && !identifier) {
+        return str.replace(/[\ud800-\udbff][\udc00-\udfff]|[\u0000-\u001f\u007f-\uffff]/g, function(ch) {
```

Comment:
I think I'm going to switch this off for now until we have enough evidence of browser support. Yes, this branch is supposed to support es6, but our output still has to be accepted by other ecma implementations like browsers.